docs/proposals/remote-artifacts.md

```diff
@@ -0,0 +1,319 @@
+can extend the API to support arbitrary configuration.
+
+### Single vs. Multiple Git Repositories
+
+For the initial implementation, we will only support one git repository that is specified in
+`.spec.source`. Remote artifacts will be specified n `.spec.sources` only.
+
+We will address supporting multipe Git repositories in a future enhancement proposal. This will
+involve re-defining `/workspace/source` location and the `$workspace` placeholder.
+
+## Steps and Helper
+
+Artifacts are requirements for the build process, thus downloading and preparing artifacts must
+happen before the build process starts. To achieve the objective, the operator will generate a new
+task step to download the artifacts.
+
+The download may happen using existing open-source software, more specifically `wget`. The
+container image which contains this software will be specified via environment variable, and when
+not informed will use a default value.
+
+Therefore, the `buildrun` controller needs to generate a `TaskRun` step, having a `wget` commands
+plus arguments, to download the specified artifacts.
+
+The step generated by the operator will look like the following:
+
+```yml
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: artifact-download
+spec:
+  steps:
+    - name: artifact-download
+      image: busybox:latest
+      command:
+        - /bin/bash
+      args:
+        - -c
+        - >
+          wget --output="/workspace/source/classpath/main.jar" https://nexus.company.com/app/main.jar && \
+          chown 1000:1001 /workspace/source/classpath/main.jar && \
+          chmod 0644 /workspace/source/classpath/main.jar
+```
+
+## Example Use-Case
+
+Using [Shipwright's proposed logos](https://github.com/shipwright-io/build/issues/325) as example,
+let's assume we are building a [TypeScript application](https://github.com/otaviof/typescript-ex)
+which will use the project logo, and we would like to create two different builds, one with the
+default project logo, and another with the alternative.
+
+By using remote artifacts, we can keep the separation of project source code and assets, and we
+can describe those resources as:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildSource
+metadata:
+  name: ship-logo
+spec:
+  sources:
+    - name: project-logo
+      type: http
+      url: https://user-images.githubusercontent.com/2587818/92114986-69bfb600-edfa-11ea-820e-96cdb1014f58.png
+      http:
+        path: $(workspace)/assets/images/shipwright-logo.png
+```
+
+And, the alternative logo:
+
+```yml
+---
+apiVersion: build.dev/v1alpha1
+kind: BuildSource
+metadata:
+  name: axes-logo
+spec:
+  sources:
+    - name: project-logo
+      type: http
+      url: https://user-images.githubusercontent.com/2587818/92100668-c1ebbd80-ede4-11ea-9e8a-7379c3875ea0.png
+      http:
+        path: $(workspace)/assets/images/shipwright-logo.png
+```
+
+Then, we can create the `Build` resource, as per:
+
+```yml
```

Comment:
@otaviof pls correct me if I´m wrong,  but here the example is based on the assumption that we support a new `BuildSource` CRD. If we do not have the `BuildSource` CRD in place, then we will still need to have 

- A Build instance to define the git source code
- Two BuildRuns, each one overriding the `spec.sources` for each logo type

is the above correct?

Reply:
Yes, correct @qu1queee. The example was made to illustrate both scenarios at once, but it might have been too much leaning towards the "Alternative" scenario.

Reply:
ok, thanks